include/librealsense2/dds/dds-device.cpp

```diff
@@ -36,7 +74,180 @@ class dds_device::impl
+                            if( state_type::WAIT_FOR_DEVICE_HEADER == state )
+                            {
+                                num_of_sensors = data.get< topics::device::notification::device_header_msg >()->num_of_sensors;
+                                LOG_INFO( "got DEVICE_HEADER message with " << num_of_sensors << " sensors" );
+                                state = state_type::WAIT_FOR_SENSOR_HEADER;
+                            }
+                            else
+                                LOG_ERROR( "Wrong message received, got 'DEVICE_HEADER' when the state is: " << state );
+                            break;
+                        case topics::device::notification::msg_type::SENSOR_HEADER: {
+                            if( state_type::WAIT_FOR_SENSOR_HEADER == state )
+                            {
+                                auto sensor_header = data.get< topics::device::notification::sensor_header_msg >();
+                                LOG_INFO( "got SENSOR_HEADER message for sensor: " << sensor_header->name << " of type: "
+                                          << ( sensor_header->type == topics::device::notification::sensor_header_msg::sensor_type::VIDEO_SENSOR
+                                                   ? "VIDEO_SENSOR"
+                                                   : "MOTION_SENSOR" ) );
+
+                                sensor_index_to_name.emplace( sensor_header->index, sensor_header->name );
+                                state = state_type::WAIT_FOR_PROFILES;
+                            }
+                            else
+                                LOG_ERROR( "Wrong message received, got 'SENSOR_HEADER' when the state is: " << state );
+                        }
+                        break;
+                        case topics::device::notification::msg_type::VIDEO_STREAM_PROFILES:
+                            if( state_type::WAIT_FOR_PROFILES == state )
+                            {
+                                auto video_stream_profiles = data.get<topics::device::notification::video_stream_profiles_msg >();
+                                
+                                LOG_INFO( "got VIDEO_STREAM_PROFILES message" );
+                                
+                                // TODO: Try to save the "emplace" profiles copy 
+                                topics::device::notification::video_stream_profiles_msg msg;
+                                msg.dds_sensor_index = video_stream_profiles->dds_sensor_index;
+                                msg.num_of_profiles = video_stream_profiles->num_of_profiles;
+                                for( int i = 0; i < video_stream_profiles->num_of_profiles; ++i )
+                                    msg.profiles[i] = video_stream_profiles->profiles[i];
+
+                                sensor_to_video_profiles.emplace( sensor_index_to_name.at(video_stream_profiles->dds_sensor_index), std::move( msg ) );
```

Comment:
We need to check everything, in case the profiles are invalid -- in this case, the `at()` will throw if the sensor index is bad.

Reply:
I will add protection for the at, since I write both sides I ignored it currently.
But you are right, better protect it.

Reply:
Done